src/box/sql/expr.c

```diff
@@ -3850,10 +3850,12 @@ sqlExprCodeTarget(Parse * pParse, Expr * pExpr, int target)
 		}
 	case TK_ID:
 		assert(pParse->vdbe_field_ref_reg > 0);
+		const char *name = pExpr->u.zToken;
 		int reg = pParse->vdbe_field_ref_reg;
-		sqlVdbeAddOp4(v, OP_Fetch, reg, 0, target,
-			      sqlDbStrDup(pParse->db, pExpr->u.zToken),
-			      P4_DYNAMIC);
+		if (sql_vdbe_fetch_by_name(v, name, reg, target) != 0) {
```

Comment:
That is a strange name. The only other function having `sql_vdbe` prefix is `sql_vdbe_prepare`. So the name isn't consistent with other functions which emit code. It also looks like it fetches something from vdbe object by its name, which isn't true. It just appends a newly generated operation to the machine. I suggest to make its naming consistent with other code-emission functions.

Reply:
Thank you for the review! I have dropped this function as per your suggestions below.